### DISCUSSION TITLE
Tool tray

### DISCUSSION THREAD

**Discussion by florence-wolfe** [jon-harper/OmniBox#49] [Ideas]
I haven't fully fleshed out this idea yet, so I'd like to start a discussion around it.

As far as I can see, there are opportunities to leverage some of the empty space in the omnibox to add in a tool tray.
Here are a few ideas off the top of my head:

- The entire base of the case is 2-3" tall and could be modified to fit a tray that could slide in and out. This would be incredibly handy to store convenient tools that would be necessary to make modifications to either the case, internal components, etc.
- The CPU slot could have a small tray if a CPU is not being used to make better use of the space instead of the blank slot cover
- A new lid could be designed to accommodate a tool drawer system on top of the case
  - example - [Fast-Printing Modular Drawer System (Vase Mode)](https://www.printables.com/model/139570-fast-printing-modular-drawer-system-vase-mode)
- The main body could have holes for "hooks" to behave similarly to pegboards to allow for hooking all sorts of additional tools and components to the main body of the case

I'd be really curious to hear what other people think about this!

**Comment by jon-harper** [comment]
First of all, thanks for sharing! I'll break down by location:

## Add to the base

> The entire base of the case is 2-3" tall and could be modified to fit a tray that could slide in and out.

The base itself is mostly occupied by the power supply, unfortunately.

*However* the 0.9.9 release includes a base extension that sits between the base and the (new) bottom panel. It's highlighted below:

![image](https://user-images.githubusercontent.com/11852387/198037403-3b17ddbb-093a-413a-ba82-03a68a8a63e0.png)

What I'm thinking is to make drawer guides into a base extension.

## CPU Tray Drawer

> The CPU slot could have a small tray if a CPU is not being used to make better use of the space instead of the blank slot cover

Consider this on the "sooner than later" to-do list for certain. 0.9.9 includes two CPU bays, so that's twice the fun.

## Stacking Lids

> A new lid could be designed to accommodate a tool drawer system on top of the case 

I like it in theory, but removing the lid would require removing the drawers, too. That'll need some thought to get right. Handles to lift it? Or maybe make it compatible with the drawer system that you shared?

## Hooks and Pegs

> The main body could have holes for "hooks" to behave similarly to pegboards to allow for hooking all sorts of additional tools and components to the main body of the case

I'm less initially enthusiastic about this idea...

- Building holes into the side of the case would impact existing user concerns about dust intrusion.
- Hooks would add print time/filament.

...but I propose an alternative:

OTOH, the new side panels in 0.9.9 (a fancy name for an unused CPU tray cover) are meant to allow customisations like this. I'm looking at adding more of them to the case in the follow-on to 0.9.9 so that every case has at least 4. Making a side panel with a hook or pegboard holes would be pretty easy. Thoughts?

------

Thanks again. Let me know what you think about my feedback.

**Comment by jon-harper** [comment]
Hey Frank! I woke up this morning and felt like making a storage tray...

[Here's the commit.](https://github.com/jon-harper/OmniBox/commit/990aa84ab318b726b4554c04eba601df6b2f2891) 
Feedback definitely welcome. :smiley:

![image](https://user-images.githubusercontent.com/11852387/218312566-bd6430b5-6475-4ac6-a977-7b177dd3332d.png)
![image](https://user-images.githubusercontent.com/11852387/218312571-9c9d263a-cac5-47c1-8742-ad2381d7d62b.png)

**Reply by jon-harper** [reply]
I'm not settled on the handle, btw.

**Reply by Otann** [reply]
I suppose if you make it hollow from left to right, it could make it more grippable

**Comment by jon-harper** [comment]
![image](https://github.com/jon-harper/OmniBox/assets/11852387/aa038da7-cdcc-481b-af03-f0ca573ab19a)

This is a little shorter than the last one. I found the top too close to the top lip of the CPU tray and required some force to remove. I also modified the handle for grip.

It's in the `0.9.10-beta` branch here: https://github.com/jon-harper/OmniBox/tree/0.9.10-beta/Trays/CPU/Storage%20Tray
